test/test_sparql_converter.js

```diff
@@ -0,0 +1,107 @@
+        "https://query.wikidata.org/sparql?query=%0ASELECT%20distinct%20%3Fv1%20%3Fv1Label%20%3Fv2Label%20%3Fv3Label%20WHERE%7B%0A%0A%0A%3Fv1%20wdt%3AP647%20wd%3AQ162990.%0A%3Fv1%20wdt%3AP166%20wd%3AQ222047.%0A%0A%0A%0A%0A%0ASERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%22.%20%7D%0A%0A%7D%0A%0Alimit%2010%0A",
+        "https://query.wikidata.org/sparql?query=%0ASELECT%20distinct%20%3Fv1%20%3Fv1Label%20%3Fv2Label%20%3Fv3Label%20WHERE%7B%0A%0A%0A%3Fv1%20wdt%3AP54%20wd%3AQ121783.%0A%3Fv1%20wdt%3AP54%20wd%3AQ157376.%0A%0A%0A%0A%0A%0ASERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%22.%20%7D%0A%0A%7D%0A%0Alimit%2010%0A",
+        "https://query.wikidata.org/sparql?query=%0ASELECT%20distinct%20%3Fv1%20%3Fv1Label%20%3Fv2Label%20%3Fv3Label%20WHERE%7B%0A%0A%0A%3Fv1%20wdt%3AP286%20wd%3AQ523630.%0A%0A%3Fv2%20wdt%3AP647%20%3Fv1.%0A%0A%3Fv2%20wdt%3AP569%20%3Fv3.%0A%0A%0A%0ASERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%22.%20%7D%0A%0A%7D%0A%0Alimit%2010%0A",
+        "https://query.wikidata.org/sparql?query=%0ASELECT%20distinct%20%3Fv1%20%3Fv1Label%20%3Fv2Label%20%3Fv3Label%20WHERE%7B%0A%0A%3Fitem1%20%3Flabel%20'Curry'%40en.%0A%0A%3Fv1%20wdt%3AP734%20%3Fitem1.%0A%3Fv1%20wdt%3AP641%20wd%3AQ5372.%0A%0A%0A%0A%0A%0ASERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%22.%20%7D%0A%0A%7D%0A%0Alimit%2010%0A",
+        "https://query.wikidata.org/sparql?query=%0ASELECT%20distinct%20%3Fv1%20%3Fv1Label%20%3Fv2Label%20%3Fv3Label%20WHERE%7B%0A%0A%0A%3Fv1%20wdt%3AP647%20wd%3AQ157376.%0A%0A%0A%0A%3Fv1%20wdt%3AP569%20%3Fcounter.%0A%0A%0ASERVICE%20wikibase%3Alabel%20%7B%20bd%3AserviceParam%20wikibase%3Alanguage%20%22en%22.%20%7D%0A%0A%7D%0AORDER%20BY%20desc(%3Fcounter)%0Alimit%2010%0A",
+    ];
+
+    Promise.all(
+        code.map((code) => {
+            let promise = new Promise((resolve, reject) => {
+                code = code.trim();
+                AppGrammar.parseAndTypecheck(code, _schemaRetriever).then(
+                    async (program) => {
+                        //convert from ast to sparql
+                        const sparqlQuery = await SparqlConverter.toSparql(
+                            program
+                        );
+
+                        const queryURL =
+                            "https://query.wikidata.org/sparql" +
+                            "?query=" +
+                            encodeURIComponent(sparqlQuery[0]);
+                        resolve(queryURL);
+                    }
+                );
+            });
+            return promise;
```

Comment:
This promise code seems to be needlessly convoluted. Why do you need to create a `new Promise`?